src/librbd/internal.cc

```diff
@@ -68,6 +68,11 @@ namespace librbd {
     return image_name + RBD_SUFFIX;
   }
 
+  const string object_map_name(const string &image_id)
+  {
+    return RBD_OBJECT_MAP_PREFIX + image_id;
```

Comment:
You might have a race condition if you use a per-snapshot object map -- unless you still snapshot the main object map and copy the snapshot contents to a new object.